make/e2e-setup.mk

```diff
@@ -45,7 +45,9 @@ IMAGE_projectcontour_arm64 := ghcr.io/projectcontour/contour:v1.25.2@sha256:abbb
 PEBBLE_COMMIT = ba5f81dd80fa870cbc19326f2d5a46f45f0b5ee3
 
 LOCALIMAGE_pebble := local/pebble:local
-LOCALIMAGE_vaultretagged := local/vault:local
+# Export this variable so that the image reference can be used by the Vault
+# addon in the E2E tests
+export LOCALIMAGE_vaultretagged := local/vault:local
```

Comment:
Could we maybe introduce a new variable that equals LOCALIMAGE_vaultretagged but is exported (eg. `export E2E_VAULT_IMAGE`)?

Reply:
Also, this breaks the flow where you would "make -j e2e-setup" and go debug afterwards (without using make in 2nd step).

Reply:
> Its quite confusing that this one variable is exported and the other two LOCALIMAGE variables are not.

Yeah, I agree. I'll try and change it.

> Also, this breaks the flow where you would "make -j e2e-setup" and go debug afterwards (without using make in 2nd step)

I don't think **this** change breaks that flow. 
That flow was broken by my previous PR https://github.com/cert-manager/cert-manager/pull/6387

In that flow, you could still use `go test ./test/e2e/...` in the second step. 
But it would install the Vault helm chart using the docker.io images rather than those that had been loaded into the kind cluster by `make e2e-setup.
Which isn't ideal, but it would only break if you're trying to run the tests offline.